tests/unit/geo.tcl

```diff
@@ -456,4 +589,92 @@ start_server {tags {"geo"}} {
+            set height_m [expr {[randomInt 10000]+10}]
+            set lat_delta [geo_raddeg [expr {$height_m/2/6372797.560856}]]
+            set long_delta_top [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat+$lat_delta}]])}]]
+            set long_delta_middle [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad $search_lat])}]]
+            set long_delta_bottom [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat-$lat_delta}]])}]]
+
+            # Total of 8 points are generated, which are located at each vertex and the center of each side
+            set points(0) [expr {$search_lon-$long_delta_top}]
+            set points(1) [expr {$search_lat+$lat_delta}]
+            set points(2) [expr {$search_lon+$long_delta_top}]
+            set points(3) [expr {$search_lat+$lat_delta}]
+            set points(4) [expr {$search_lon+$long_delta_bottom}]
+            set points(5) [expr {$search_lat-$lat_delta}]
+            set points(6) [expr {$search_lon-$long_delta_bottom}]
+            set points(7) [expr {$search_lat-$lat_delta}]
+            set points(8) $search_lon
+            set points(9) [expr {$search_lat+$lat_delta}]
+            set points(10) $search_lon
+            set points(11) [expr {$search_lat-$lat_delta}]
+            set points(12) [expr {$search_lon+$long_delta_middle}]
+            set points(13) $search_lat
+            set points(14) [expr {$search_lon-$long_delta_middle}]
+            set points(15) $search_lat
+
+            lappend debuginfo "Search area: geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_m $height_m m"
+            set tcl_result {}
+            for {set i 0} {$i < 16} {incr i 2} {
+                set x $points($i)
+                set y $points([expr {$i+1}])
+                r geoadd mypoints $x $y place:$i
+                lappend tcl_result "place:$i"
+                lappend debuginfo "geoadd mypoints $x $y place:$i"
+            }
+
+            set res2 [lsort $tcl_result]
+
+            # make the box larger by two meter in each direction to put the coordinate slightly inside the box.
+            set height_new [expr {$height_m+4}]
+            set width_new [expr {$width_m+4}]
+            set res [lsort [r geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_new $height_new m]]
+            if {$res != $res2} {
+                set diff [compare_lists $res $res2]
+                lappend debuginfo "diff: $diff"
+                fail "place should be found, debuginfo: $debuginfo, height_new: $height_new width_new: $width_new"
+            }
+
+            # Only width reduced, the four corners and the midpoint on the left and right will not be searched
+            set width_new [expr {$width_m-4}]
+            set res [lsort [r geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_new $height_m m]]
+            if {$res != {place:10 place:8}} {
```

Comment:
doesn't this make the test possibly flaky? i.e. the two points that are found are right on the border, due to inaccuracy issues they may be found time times and not found other times?
so we know the corners and top and right midpoints will not be found, but we're not sure about the left and right.

maybe one way to keep the current `if {$res != {place:10 place:8}} {` and make it stable is change the two tests to increase one dimension and decrease the other.
in this case we'll have 3 tests:
1. increase both width and height (all points are found)
2. increase width and decrease height (corners and top, bottom edges are found. only left and right)
3. decrease the width and increase the height (corners and left, right edges are found. only top and bottom)
4. if we want we can decrease both and make sure nothing is found but i don't think this test has value.

Reply:
sorry.. ignore that suggestion of increasing width and decreasing height (the corners will not be found).
let's just consider the accuracy of whether place:10 and place:8 must be found (think sometimes they won't be found, and that's not an error)